Cookbook/Proposals/DecoupleSnapshotTestsFromUnitTests.md

```diff
@@ -0,0 +1,31 @@
+# Decouple Snapshot tests from unit tests
+* Author(s): Adam Borek
+
+## Introduction
+The purpose of this proposal is to decouple snapshot tests from unit tests to be able to run only unit-tests which takes much less time to execute than snapshot tests.
+
+The goal of this proposal is to make it possible to run only unit-tests **locally** in Xcode. Improving CI run time is not a goal of this proposal.
+
+## Motivation
+Right now when we run all unit-tests (by pressing cmd+U) we also run snapshot tests. Right now it takes **14 minutes** to run all unit+snapshot tests which is very long. Those 14 minutes doesn't include the build time 😱.
+
+Unit-tests should run as quickly as possible so we can have a quick feedback about stability of our changes.\
+\
+Ideally, it would be nice to be able to run only unit-tests when we don't work on UI. Unit-tests in separation take about ~2% of the execution time.
+
+## Proposed solution
+Decouple snapshot tests from unit test by creating `*SnapshotTests` targets in every framework which has snapshots.
+
+We would also need to create a new scheme like `BabylonUniTestsOnly` which would run only tests targets without snapshot tests targets.
+
+`Babylon` scheme would still remain configuration to run all tests (unit + snapshots).
+
+## Impact on existing codebase
+
+1. We would have more tests targets
+2. Once we create any test file we would need to be cautious to add it to the correct test target. Unit tests into `*Tests`, snapshot tests into `*SnapshotTests`
+3. We would need to have additional scheme like `BabylonUnitTestsOnly` which would run only unit-tests on `cmd+u`.
+
+## Alternatives considered
+
+- **Xcode's 11 Test plans** - Unfortunately, I haven't found a way to select test files per configuration. I'm afraid we cannot use test plans.
```

Comment:
can you elaborate on that? Test plans can be created from scheme, so if we create BabylonUnitTests scheme it can be as well converted to a test plan. And schemes can have multiple test plans assigned to them. 

https://shashikantjagtap.net/wwdc19-getting-started-with-test-plan-for-xctest/

> You can run the test using the Xcode ->Product -> Test for your scheme as usual which executes the attached test plan for the scheme. In our case, it will execute both smoke and regression test configured in the functional test plan.

It's not necessary that we should use test plans as they might not bring enough value for us, but the conclusion that we can't use them at all seems premature

The other thing we would need to explore in regards of test plans is using them with fastlane

Reply:
Yes, I used a wrong word here. We can use test plans. TestPlans just won't help in context of this proposal. We would still need 2 targets, one for unit and one for snapshot tests.

I'll change the explanation above to be more clear.

Reply:
don't they allow to include different test cases without defining different targets?

Reply:
It's tricky. They do, however, if you add a new test file then they are automatically enabled in every TestPlan which uses the target. That means, if I add a `NewFeatureSnapshotTest.swift` it would be enabled in both TestPlans for unit-tests and snapshot-tests.

Reply:
Did I hear someone saying "Danger"?...
But ok, got your point. This can be included to the doc imo